### PR TITLE
Add ApiOptions overloads to I(Observable)RepositoriesClient

### LINKED ISSUE #1171: Add ApiOption overloads to methods on IRepositoriesClient
As part of completing #760 a number of overloads need to be added to the client interfaces.

As an example, this has been added to [`IReleasesClient`](https://github.com/octokit/octokit.net/commit/03ea83148995711fc47148489e2123e65f33bd82)and [`IObservableReleasesClient`](https://github.com/octokit/octokit.net/commit/aa3ce0db9210b778414e290af7f17c4a646bdeec).

The checklist for porting each method is:

- Add an overload for this method on `IRepositoriesClient` with a new parameter `ApiOptions`
- Implement the new method on the class which implements `IRepositoriesClient`
- Add an overload for this method on `IObservableRepositoriesClient` with a new parameter `ApiOptions`
- Implement the new method on the class which implements `IObservableRepositoriesClient`

This needs to be done for each of these methods:

- [ ] `Task<IReadOnlyList<Branch>> GetAllBranches(String owner, String name)`
- [ ] `Task<IReadOnlyList<RepositoryContributor>> GetAllContributors(String owner, String name)`
- [ ] `Task<IReadOnlyList<RepositoryContributor>> GetAllContributors(String owner, String name, Boolean includeAnonymous)`
- [ ] `Task<IReadOnlyList<Repository>> GetAllForCurrent()`
- [ ] `Task<IReadOnlyList<Repository>> GetAllForCurrent(RepositoryRequest request)`
- [ ] `Task<IReadOnlyList<Repository>> GetAllForOrg(String organization)`
- [ ] `Task<IReadOnlyList<Repository>> GetAllForUser(String login)`
- [ ] `Task<IReadOnlyList<RepositoryLanguage>> GetAllLanguages(String owner, String name)`
- [ ] `Task<IReadOnlyList<Repository>> GetAllPublic()`
- [ ] `Task<IReadOnlyList<Repository>> GetAllPublic(PublicRepositoryRequest request)`
- [ ] `Task<IReadOnlyList<RepositoryTag>> GetAllTags(String owner, String name)`
- [ ] `Task<IReadOnlyList<Team>> GetAllTeams(String owner, String name)`

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #1171 

For each of these entries:

 - add overload to both clients
 - implement
 - update impacted unit tests
 - add new integration tests wherever possible

- [x] `Task<IReadOnlyList<Branch>> GetAllBranches(String owner, String name)`
- [x] `Task<IReadOnlyList<RepositoryContributor>> GetAllContributors(String owner, String name)`
- [x] `Task<IReadOnlyList<RepositoryContributor>> GetAllContributors(String owner, String name, Boolean includeAnonymous)`
- [x] `Task<IReadOnlyList<Repository>> GetAllForCurrent()`
- [x] `Task<IReadOnlyList<Repository>> GetAllForCurrent(RepositoryRequest request)`
- [x] `Task<IReadOnlyList<Repository>> GetAllForOrg(String organization)`
- [x] `Task<IReadOnlyList<Repository>> GetAllForUser(String login)`
- ~~[ ] `Task<IReadOnlyList<RepositoryLanguage>> GetAllLanguages(String owner, String name)`~~
- ~~[ ] `Task<IReadOnlyList<Repository>> GetAllPublic()`~~ moved to #1340
- ~~[ ] `Task<IReadOnlyList<Repository>> GetAllPublic(PublicRepositoryRequest request)`~~ moved to #1340
- [x] `Task<IReadOnlyList<RepositoryTag>> GetAllTags(String owner, String name)`
- [x] `Task<IReadOnlyList<Team>> GetAllTeams(String owner, String name)`

### CODE REVIEW

**Comment by ryangribble** [comment]
there's some weird/inconsistent test failures here... is it a similar case to what occured in #1169 where `Args.ApiOptions` is used instead of `ApiOptions.None` causing issues with parallel running unit tests?

Quote from @shiftkey 
> So I don't know the full answer here for the runtime quirks when running tests but I can at least recommend we don't use Args.ApiOptions (the wrapper around NSubstitute's assert helper Arg.Any<ApiOption>()) unless you're asserting, instead of ApiOptions.None (which is the Octokit model).
> 
> Not sure if we need to do anything else here, but something to be mindful of...

File: Octokit.Tests/Clients/RepositoriesClientTests.cs [hunk 1]
```diff
@@ -437,21 +442,27 @@ public void RequestsTheCorrectUrlAndReturnsRepositories()
                 var reposEndpoint = new RepositoriesClient(Substitute.For<IApiConnection>());
 
                 await Assert.ThrowsAsync<ArgumentNullException>(() => reposEndpoint.GetAllForUser(null));
+                await Assert.ThrowsAsync<ArgumentException>(() => reposEndpoint.GetAllForUser(""));
+
+                await Assert.ThrowsAsync<ArgumentNullException>(() => reposEndpoint.GetAllForUser(null, Args.ApiOptions));
```
Comment (dampir) [inline Octokit.Tests/Clients/RepositoriesClientTests.cs]: ApiOptions.None have to be used here 👍

File: Octokit.Tests/Clients/RepositoriesClientTests.cs [hunk 2]
```diff
@@ -437,21 +442,27 @@ public void RequestsTheCorrectUrlAndReturnsRepositories()
...
+                await Assert.ThrowsAsync<ArgumentNullException>(() => reposEndpoint.GetAllForUser("user", null));
+
+                await Assert.ThrowsAsync<ArgumentException>(() => reposEndpoint.GetAllForUser("", Args.ApiOptions));
```
Comment (dampir) [inline Octokit.Tests/Clients/RepositoriesClientTests.cs]: The same as previous remark 👍

**Comment by dampir** [comment]
@ryangribble yep, you are right, I've found 2 places where Args.ApiOptions used instead of ApiOptions.None 👍

File: Octokit.Tests/Clients/RepositoriesClientTests.cs (hunk 1 above)
Comment (shiftkey) [inline Octokit.Tests/Clients/RepositoriesClientTests.cs]: Nice pickup @dampir - this was doing my head in yesterday!

**Comment by shiftkey** [comment]
Alright, ready for review _finally_...

**Comment by shiftkey** [comment]
I'd like to run through the integration tests today and get this stuff ready for release. Are we cool with merging this?

**Pull request merged**